tests/integration.rs

```diff
@@ -1912,6 +1912,122 @@ mod tests {
+            let mut child = GuestCommand::new(&guest)
+                .args(&["--cpus", "boot=1"])
+                .args(&["--memory", "size=512M,file=/dev/shm"])
+                .args(&["--kernel", kernel_path.to_str().unwrap()])
+                .default_disks()
+                .default_net()
+                .args(&["--cmdline", CLEAR_KERNEL_CMDLINE])
+                .args(&["--api-socket", &api_socket])
+                .spawn()
+                .unwrap();
+
+            thread::sleep(std::time::Duration::new(20, 0));
+
+            // Add fs to the VM
+            aver!(
+                tb,
+                remote_command(
+                    &api_socket,
+                    "add-fs",
+                    Some(&format!(
+                        "tag=myfs,sock={},num_queues=1,queue_size=1024,dax={}{}",
+                        virtiofsd_socket_path, dax_vmm_param, cache_size_vmm_param
+                    )),
+                )
+            );
+
+            thread::sleep(std::time::Duration::new(10, 0));
+
+            // Mount shared directory through virtio_fs filesystem
+            let mount_cmd = format!(
+                "mkdir -p mount_dir && \
+                 sudo mount -t virtiofs {} myfs mount_dir/ 2>&1 && \
+                 echo ok",
+                dax_mount_param
+            );
+            aver_eq!(
+                tb,
+                guest.ssh_command(&mount_cmd).unwrap_or_default().trim(),
+                "ok"
+            );
+
+            // Check the cache size is the expected one.
+            // With virtio-mmio the cache doesn't appear in /proc/iomem
+            #[cfg(not(feature = "mmio"))]
+            aver_eq!(
+                tb,
+                guest
+                    .valid_virtio_fs_cache_size(dax, cache_size)
+                    .unwrap_or_default(),
+                true
+            );
+            // Check file1 exists and its content is "foo"
+            aver_eq!(
+                tb,
+                guest
+                    .ssh_command("cat mount_dir/file1")
+                    .unwrap_or_default()
+                    .trim(),
+                "foo"
+            );
+
+            // Ensure the VM successfully reboots. The virtio-fs will not be
+            // automatically recreated since virtiofsd exits on disconnect.
+            guest.ssh_command("sudo reboot").unwrap_or_default();
+
+            thread::sleep(std::time::Duration::new(20, 0));
+            let reboot_count = guest
+                .ssh_command("sudo journalctl | grep -c -- \"-- Reboot --\"")
+                .unwrap_or_default()
+                .trim()
+                .parse::<u32>()
+                .unwrap_or_default();
+            aver_eq!(tb, reboot_count, 1);
+
+            let _ = child.kill();
+            let _ = daemon_child.kill();
+            let _ = child.wait();
+            let _ = daemon_child.wait();
+            Ok(())
+        });
+    }
+
```

Comment:
You need to edit the commit title and commit message to indicate you're adding integration tests.

Reply:
The tests fail, so I pushed it as a temporary commit. Once the issue is fixed I'll amend the commit.

Reply:
Ok so about the failing test, is it working locally but not with the CI?

Have you tried to rebase on latest master as suggested by @rbradford since master now includes #1027 ?

Reply:
Fail locally, the ci didn't run for some reason so I can't check. I tried again with the #1027 patch before it was merged, I'll try again with a rebase on master.

Reply:
Oh it fails locally, then this is gonna be easier to debug. Let me know if you have made some progress on it, otherwise I'll try to help with it :)